lib/scanner.js

```diff
@@ -31,29 +33,34 @@ class Scanner {
 
   consumeWhitespace () {
     while (isWhitespace(this.peek())) {
-      this.pos++
+      this.pos.column++
+      if (isNewline(this.peek())) {
```

Comment:
Could this logic be moved into `next()`, I've only been calling `consumeWhitespace` for nodes where we explicitly indicate that they consume whitespace, e.g.,

`<summary-sep>  ::= "!"? ":" *<whitespace>`

Also, it's worth noting that `\r` and `\n` currently aren't checked by `isWhitespace`, which is capturing all whitespace except newline characters (perhaps we should add a comment to that effect?).

Reply:
We can move that to next, but to keep the positioning in sync, we have to use `.next` in `.consumeWhitespace`. I'll do that!